packages/plugin-ext/src/plugin/tasks/tasks.ts

```diff
@@ -165,52 +195,67 @@ export class TasksExtImpl implements TasksExt {
     }
 
     $provideTasks(handle: number, token: theia.CancellationToken): Promise<TaskDto[] | undefined> {
-        const adapter = this.adaptersMap.get(handle);
-        if (adapter) {
-            return adapter.provideTasks(token).then(tasks => {
+        const provider = this.providersByHandle.get(handle);
+        let addedExecutions = 0;
+        if (provider) {
+            return Promise.resolve(provider.provideTasks(token)).then(tasks => {
                 if (tasks) {
-                    for (const task of tasks) {
-                        if (task.taskType === 'customExecution') {
-                            task.executionId = this.addCustomExecution(task.callback);
-                            task.callback = undefined;
+                    return tasks.map(task => {
+                        const dto = converter.fromTask(task);
+                        if (dto && CustomExecution.is(task.execution!)) {
+                            dto.executionId = this.addProvidedCustomExecution(task.execution);
+                            addedExecutions++;
                         }
-                    }
+                        return dto;
+                    }).filter(task => !!task).map(task => task!);
```

Comment:
You can avoid the second `.map` here with something like this to help TS understand what you mean:
```typescript
.filter((task): task is TaskDto => !!task)
```